llama-index-integrations/workflows/llama-index-workflows-temporal/llama_index/workflows/temporal/TemporalWorkflow.py

```diff
@@ -0,0 +1,311 @@
+                    )
+                    print(f"maybe event: {maybe_event}")
+                    events = []
+                    if isinstance(maybe_event, Event):
+                        events.append(maybe_event)
+                    return events
+
+                coroutines.append(exec(activity))
+        return coroutines
+
+    @workflow.query
+    async def get_processed_events(self, msg: WorkflowContextQueryProcessedEvents) -> List[Event]:
+        return self._processed_events
+
+    @workflow.signal
+    def send_event(self, msg: WorkflowContextSendEvent) -> None:
+        self._pending_events.append(msg.event)
+
+    @workflow.query
+    async def get_result(self) -> RunResultT:
+        return self._result
+
+    @workflow.query
+    async def get_data(self, msg: WorkflowContextGet) -> Any:
+        return self._data[msg.key]
+
+    @workflow.update
+    async def set_data(self, msg: WorkflowContextSet) -> None:
+        self._data[msg.key] = msg.value
+
+
+
+def _get_full_path(ev_type: Type[Event]) -> str:
+    qualified_module = ev_type.__module__.replace("__temporal_main__", "__main__")
+    return f"{qualified_module}.{ev_type.__name__}"
+
+
+class TemporalWorkflowHandle(asyncio.Future[RunResultT]):
+
+    pass
+
+
+class TemporalWorkflowBuilder():
+    def __init__(self, client: Client, temporal_workflow: Type[LlamaIndexTemporalWorkflow], instance: Workflow, register_worker: bool = True):
+        self._client = client
+        self._wf = instance
+        self._temporal_workflow = temporal_workflow
+        steps = NamedCallable.from_dict(self._wf._get_steps())
+        self.activites = _build_activities(steps)
+        self._worker = asyncio.create_task(self._create_worker()) if register_worker else None
+
+    async def _create_worker(self):
+        pass
+        # async with Worker(
+        #     self._client,
+        #     task_queue=self._wf.__class__.__qualname__,
+        #     workflows=[self._temporal_workflow],
+        #     activities=[x.callable for x in self.activites]
+        # ):
+        #     print("created worker")
+        #     await asyncio.Future()
+
+
+    async def run(
+        self,
+        start_event: Optional[StartEvent] = None,
+        **kwargs: Any,
+    ) -> TemporalWorkflowHandle:
+        print(f"running workflow {self._wf.__class__.__qualname__}")
+        print(f"registering activities {[x.unique_name for x in self.activites]}")
+        async with Worker(
+            self._client,
+            task_queue=self._wf.__class__.__qualname__,
+            workflows=[self._temporal_workflow],
+            activities=[x.callable for x in self.activites]
+        ):
+            result = await self._client.execute_workflow(
+                self._temporal_workflow.run, start_event or self._wf._get_start_event_instance(None, **kwargs),
+                id=uuid.uuid4().hex,
+                task_queue=self._wf.__class__.__qualname__,
+            )
+            return result
```

Comment:
this needs to expose the context/streaming interface still, and no actual response is returned yet